tutorials/btp-integration-suite-execute-batch-payload/btp-integration-suite-execute-batch-payload.md

```diff
@@ -0,0 +1,377 @@
+We will replicate the above Atomic scenario with a Groovy Script.
+
+
+1. Choose **Atomic Batch Mode** and add the following script in Groovy Script as explained in Step 8.
+
+    Here we have modified the third and fourth record of the original Groovy Script.
+
+    ```
+        import com.sap.gateway.ip.core.customdev.util.Message;
+        import java.util.HashMap;
+        import java.util.Arrays;
+
+        //Insert script
+        def Message processData(Message message) {
+
+            //Headers
+            List paramList = new ArrayList();
+            paramList.add(Arrays.asList(108, 'test', 2021-01-01,1,1,'test@gmail.com'));
+            paramList.add(Arrays.asList(107, 'test', 2021-01-01,1,1,'test@gmail.com'));
+            paramList.add(Arrays.asList(112, 'test112', 2021-01-01,1,1,'test@gmail.com'));
+            paramList.add(Arrays.asList(106, 'test106', 2021-01-01,1,1,'test@gmail.com'));
+            message.setHeader("CamelSqlParameters",paramList);
+
+            //Body
+            message.setBody("INSERT INTO testjdbcsample (emp_id,emp_name,join_date,a1,a2,email) VALUES(?,?,?,?,?,?)");
+            return message;
+        }
+    ```  
+2. **Send** a blank query in Postman to send the payload which we have entered via Groovy Script.
+
+3. Check the error from MPL.
+
+    <!-- border -->![Link text e.g., Destination screen](Groovy-Atomic-MPL.png)
+
+4. Remove **Groovy Script**, switch off the **Batch Mode** and **Deploy** the integration flow. Then, run the following query from Postman to check the database updates, if any:
+
+`select * from testjdbcsample`
+
+  <!-- border -->![Link text e.g., Destination screen](Groovy-Atomic-Error.png)
+
+This proves the Atomic mode functionality as the database table is not updated. It is the same as it was after Step 8. The new record didn't insert because there were duplicate records.
+
+
+### Non-Atomic Batch Operation with Groovy Script
+
+1. Choose **Non-Atomic Batch Mode** and add the following script in Groovy Script as explained in Step 8.
+
+    ```
+    import com.sap.gateway.ip.core.customdev.util.Message;
+    import java.util.HashMap;
+    import java.util.Arrays;
+
+    //Insert script
+    def Message processData(Message message) {
+
+        //Headers
+        List paramList = new ArrayList();
+        paramList.add(Arrays.asList(108, 'test', 2021-01-01,1,1,'test@gmail.com'));
+        paramList.add(Arrays.asList(107, 'test', 2021-01-01,1,1,'test@gmail.com'));
+        paramList.add(Arrays.asList(112, 'test112', 2021-01-01,1,1,'test@gmail.com'));
+        paramList.add(Arrays.asList(106, 'test106', 2021-01-01,1,1,'test@gmail.com'));
+        message.setHeader("CamelSqlParameters",paramList);
+
+        //Body
+        message.setBody("INSERT INTO testjdbcsample (emp_id,emp_name,join_date,a1,a2,email) VALUES(?,?,?,?,?,?)");
+        return message;
+    }
+    ```
+2. **Send** a blank query in Postman to send the payload which we have entered via Groovy Script.
```

Comment:
Don't say "we" when addressing the author. Say "you" instead!
Missing comma before "which"